### PR TITLE
net/haproxy: release 3.0

### PR DESCRIPTION
**Pull request by fraenki**
Main achievements in this release:

- [X] new feature: change server state and weight on-the-fly (#2213)
- [X] new feature: support runtime certificate updates (#1882, #2244)
- [x] switch to HAProxy 2.2 release series (#2092)
- [x] change **default** SSL parameters (TLSv1.2+, update ciphers, remove outdated default values; will not change existing configs)

See plugin changelog for a complete list of changes and new features.

### CODE REVIEW

File: net/haproxy/Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 PLUGIN_NAME=		haproxy
-PLUGIN_VERSION=		2.26
+PLUGIN_VERSION=		3.0
 PLUGIN_REVISION=	1
```
Comment (fichtner) [inline net/haproxy/Makefile:3]: Need to clear revision for 3.0

Comment (fraenki) [reply net/haproxy/Makefile:3]: OK :)

File: net/haproxy/src/opnsense/mvc/app/views/OPNsense/HAProxy/maintenance.volt [hunk 2]
```diff
@@ -45,10 +45,10 @@ POSSIBILITY OF SUCH DAMAGE.
...
-                        buttons += " <button type=\"button\" title=\"Set administrative state to maintenance. Disables any traffic to the server as well as any health checks.\" class=\"btn btn-xs btn-default command-set-state\" data-state=\"maint\" data-row-id=\"" + row.id + "\"><span class=\"fa fa-wrench\"></span></button>"
-                        buttons += " <button type=\"button\" title=\"Change a server's weight.\" class=\"btn btn-xs btn-default command-set-weight\" data-weight=\"" + row.weight + "\" data-row-id=\"" + row.id + "\"><span class=\"fa fa-balance-scale\"></span></button>"
+                        buttons += "<button type=\"button\" title=\"{{ lang._('Set administrative state to ready. Puts the server in normal mode.') }}\" class=\"btn btn-xs btn-default command-set-state\" data-state=\"ready\" data-row-id=\"" + row.id + "\"><span class=\"fa fa-check\"></span></button>"
```
Comment (fichtner) [inline net/haproxy/src/opnsense/mvc/app/views/OPNsense/HAProxy/maintenance.volt:48]: I'm not sure how robust lang._() is WRT properly escaping e.g. " and breaking JS worst case with translation. Let me check...

Comment (fichtner) [reply net/haproxy/src/opnsense/mvc/app/views/OPNsense/HAProxy/maintenance.volt:48]: Nevermind, we are good ;)

https://github.com/opnsense/core/blob/8158ef8a6cc903979633a021cf9365d2975457df/src/opnsense/mvc/app/library/OPNsense/Base/ViewTranslator.php#L43

**Comment by fraenki** [comment]
As stated in #2244, we're just finishing the "cert update" GUI and afterwards os-haproxy 3.0 can be merged to stable/21.1.

**Comment by fraenki** [comment]
@fichtner It's finally done, os-haproxy 3.0 is ready for release. Please merge all changes from master to stable/21.1 for the next release. One question: Would it be possible to add a more visible note to the release notes this time? There may be some breaking changes, so something like this would come to my mind:

> The HAProxy plugin was updated to version 3.0. This release marks the switch to the HAProxy 2.2 release series, which may result in incompatible changes for some users. Many new features were also added, including the possibility to update SSL certificates in runtime. These features should be considered experimental. We encourage everyone to install this version in a test environment before using it in production. As usual, please have a look at the plugin changelog and report bugs on GitHub.

Feel free to rephrase or write your own :)

**Comment by fichtner** [comment]
Ok, but my hotfixing capability is limited until end of March... 21.1.3 will be next week. It’s up to you how confident you are about 3.0. 😊

**Comment by fraenki** [comment]
@fichtner We're using it on a number of test hosts and performed several upgrade tests. No hotfixing capability is OK with this regard :)

**Comment by fichtner** [comment]
Ok done 👍

**Pull request merged**